Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs

```diff
@@ -310,139 +356,147 @@ public void TryClose(IEntity user)
             Close();
         }
 
-        private void CheckCrush()
+        public bool CanCloseByEntity(IEntity user)
         {
-            if (!Owner.TryGetComponent(out IPhysicsComponent? body))
-                return;
-
-            // Crush
-            foreach (var e in body.GetCollidingEntities(Vector2.Zero, false))
+            if (!CanCloseGeneric())
             {
-                if (!e.TryGetComponent(out StunnableComponent? stun)
-                    || !e.TryGetComponent(out IDamageableComponent? damage)
-                    || !e.TryGetComponent(out IPhysicsComponent? otherBody))
-                    continue;
-
-                var percentage = otherBody.WorldAABB.IntersectPercentage(body.WorldAABB);
-
-                if (percentage < 0.1f)
-                    continue;
-
-                damage.ChangeDamage(DamageType.Blunt, DoorCrushDamage, false, Owner);
-                stun.Paralyze(DoorStunTime);
-
-                // If we hit someone, open up after stun (opens right when stun ends)
-                Owner.SpawnTimer(TimeSpan.FromSeconds(DoorStunTime) - OpenTimeOne - OpenTimeTwo, Open);
-                break;
-            }
-        }
-
-        public bool IsHoldingPressure(float threshold = 20)
-        {
-            var atmosphereSystem = EntitySystem.Get<AtmosphereSystem>();
-
-            if (!Owner.Transform.Coordinates.TryGetTileAtmosphere(out var tileAtmos))
                 return false;
-
-            var gridAtmosphere = atmosphereSystem.GetGridAtmosphere(Owner.Transform.GridID);
-
-            var minMoles = float.MaxValue;
-            var maxMoles = 0f;
-
-            foreach (var (_, adjacent) in gridAtmosphere.GetAdjacentTiles(tileAtmos.GridIndices))
+            }
+            if (!Owner.TryGetComponent(out AccessReader? accessReader))
```

Comment:
componentdependencies

Reply:
I think this one's fine without it given doors don't need an accessreader (and saltern definitely has a few without it).